test/e2e/kubectl/kubectl.go

```diff
@@ -817,6 +817,15 @@ metadata:
 				}
 			}
 		})
+
+		ginkgo.It("should apply a CR with unknown fields for CRD with no validation schema", func() {
+			testApplyRandomCR(f)
+
+		})
+
+		ginkgo.It("should apply a CR with unknown fields for CRD with no validation schema when publish CRD openapi feature is enabled [Feature:CustomResourcePublishOpenAPI]", func() {
```

Comment:
currently version-skew testing doesn't enable any alpha feature

Reply:
why are we running the same test on line 822 and line 827, and feature-guarding one of them?

Reply:
The feature is server-side-only gates by the feature gate. So in 1.15 (when we probably promote it to beta), it will be enabled on the server side. The 1.14 kubectl will then run the skew tests.

Reply:
but... if the test on 822 always runs unconditionally, why do we need to run it again conditionally here?

Reply:
line 822 tests 
- 1.13 kubectl against 1.14 server with gate disabled
- 1.14 kubectl against 1.13 server
- 1.14 kubectl against 1.15 server
- 1.15 kubeclt against 1.14 server with gate disabled

had line 827 because I wanted to test
- 1.13 kubectl against 1.14 server with gate enabled

but it's not a blocker for publishing CRD openapi. Will drop it

Reply:
unless the `1.13 kubectl against 1.14 server with gate enabled` test is actively *excluding* tests, line 822 would test that as well, right?

Reply:
none of the [version skew tests](https://github.com/kubernetes/test-infra/blob/6289d6514c037b1502911a1beac933acb3e1bd78/config/jobs/kubernetes/sig-cli/sig-cli-config.yaml#L120) has `--runtime-config=api/all=true` set (actually a lot of them have `--ginkgo.skip=\[Feature:.+\]`), therefore `1.13 kubectl against 1.14 server with gate enabled` is currently impossible without updating test-infra

for updating test-infra, my understanding is that we have a convention of:

1. alpha tests with:
- `--ginkgo.focus=\[Feature:XXX\]`
- `--runtime-config=api/all=true`

2. non-alpha tests with:
- `--ginkgo.skip=\[Feature:.+\]`

of course we could also ignore `ginkgo.skip/focus` and just dup the jobs with `--runtime-config=api/all=true`, then line 822 will run in both cases (alpha feature enabled and disabled)